proposals/nixpkgs-security.md

```diff
@@ -0,0 +1,278 @@
+
+We will express the significance in matters of "what would a win look like".
+
+More precisely:
+
+A security team usually provides many automation for security team, maintainers and end users:
+
+- Triaging interface to ACK/mark as irrelevant/etc. vulnerabilities as they come from many pipelines, loosely matched with nixpkgs
+- Monitoring the delivery pipeline (hydra.nixos.org) to notify users or alert security team when all non-EOL distribution channels receive security updates
+- Statistics and tracking over time of vulnerabilities for research and organizational improvement
+
+In the past, Archlinux's security tracker was considered for reuse in NixOS but it is very old and tailored to Archlinux needs.
+
+Here, we propose to write a new security tracker with NixOS in mind, but in a generic way, which can be reused, not only for nixpkgs but any overlay of nixpkgs and can be deployed in organisations and for nixpkgs needs.
+
+Our main challenge is to tackle the information overload generated by the ingestion process from many security feeds, we would tackle this by providing powerful filtering capabilities linking all the available metadata from Repology and nixpkgs for example.
+
+A security.nixos.org prototype usable by Security Team would:
+
+- Reduce the intense churn on security team and foster sustainability for FOSS volunteers
+- Improve security posture of nixpkgs, especially for hard to track channels, e.g. "soon to be end of life" NixOS releases
+- Make expectations to users transparent about "when will this security vulnerability will be patched in my system" and inform them about their options, e.g. [grafting](https://guix.gnu.org/manual/en/html_node/Security-Updates.html) (`system.replaceRuntimeDependencies`) where long time before delivery are predicted based on existing data.
+
+As for boot security, NixOS recently acquired [a form of user-owned Secure Boot](https://github.com/nix-community/lanzaboote) as part of [a NLnet grant](https://nlnet.nl/project/NixOS-UEFI/).
+
+As of now, Secure Boot in nixpkgs is slowly being upstreamed as part of https://github.com/NixOS/nixpkgs/pull/231951 but the Secure Boot ecosystem is still in its infancy with systemd having a lot of features which are not necessarily completely all wired up in Linux distributions.
+
+The next stage after Secure Boot is to have measured boot which is the final piece to have a reasonable form of security in the boot of a Linux distribution.
+
+To achieve this, one can leverage `systemd-cryptenroll` tooling to tie various entities to cryptographic entities in TPM2s.
+
+The problem is that systemd semantics are not totally aligned with NixOS rollback capabilities and some footwork upstream has to be finished:
+
+- https://github.com/systemd/systemd/pull/28057
+- https://github.com/systemd/systemd/pull/28070
```

Comment:
Awesome, linking these PRs shows that work is already ongoing and a little bit of funding could put it over the finish line. The Open Source ecosystem in general also benefits, which is a plus for the chances of this proposal being accepted.